Using a Pre-Run Script with Container Executor crashes Testkube API Server

**Describe the bug**
I've defined a Test which declares a `preRunScript`. The test itself uses a container executor.
When running the test the testkube-api-server pod crashes everytime due to a Go panic.

**To Reproduce**
Steps to reproduce the behavior:
See above.

**Expected behavior**
Pre-run script runs.

**Version / Cluster**

* Which testkube version?
  API Version: v1.16.33
  Helm Chart Version: v1.16.52
* What Kubernetes cluster?
  EKS
* What Kubernetes version?
* 1\.29

**Additional context**

The error cause seems pretty obvious: we try to inspect the Docker image (probably to determine the avalable shell) and then execute the following code in `skopeop/client.go`:

```golang
if len(c.dockerAuthConfigs) != 0 {
 i := 1 + rand.Intn(len(c.dockerAuthConfigs))
 args = append(args, "--creds", c.dockerAuthConfigs[i].Username+":"+c.dockerAuthConfigs[i].Password)
}
```

The `1 +` causes an index out of range panic when accessing `c.dockerAuthConfigs[i]`. This always fails if we only have a single docker auth configuration and randomly fails if there is more than one.

Not sure what the intention was. Unfortunately the \`Inspect function doesn't seem to be covered by unit tests.

Error and stacktrace:

```
panic: runtime error: index out of range [1] with length 1 │
│ │
│ goroutine 604 [running]: │
│ github.com/kubeshop/testkube/pkg/skopeo.(*client).Inspect(0xc0017a5fb0, {0xc0016ad920, 0x2e}) │
│ /home/runner/work/testkube/testkube/pkg/skopeo/client.go:88 +0x42d │
│ github.com/kubeshop/testkube/pkg/imageinspector.(*skopeoFetcher).Fetch(0xc001be88b8?, {0x16?, 0xc0016ad920?}, {0xc00005a052?, 0xc00005a052?}, {0xc0016ad920, 0x2e}, {0xc002868dc0, 0x1, 0 │
│ x1}) │
│ /home/runner/work/testkube/testkube/pkg/imageinspector/skopeofetcher.go:24 +0x5e │
│ github.com/kubeshop/testkube/pkg/imageinspector.(*inspector).fetch(0xc0008d2640, {0x2a506c0, 0x3edc160}, {0xc00005a052, 0x1f}, {0xc0016ad920, 0x2e}, {0xc001202640, 0x1, 0x1}) │
│ /home/runner/work/testkube/testkube/pkg/imageinspector/inspector.go:55 +0x248 │
│ github.com/kubeshop/testkube/pkg/imageinspector.(*inspector).Inspect(0xc0008d2640, {0x2a506c0?, 0x3edc160?}, {0xc00005a052, 0x1f}, {0xc0016ad920, 0x2e}, {0x2612946?, 0x0?}, {0xc00120264 │
│ 0, ...}) │
│ /home/runner/work/testkube/testkube/pkg/imageinspector/inspector.go:88 +0x105 │
│ github.com/kubeshop/testkube/pkg/executor/containerexecutor.NewJobOptions(_, {_, _}, {{_, _}, {_, _}, {_, _}}, {{0xc000287500, ...}, ...}, ...) │
│ /home/runner/work/testkube/testkube/pkg/executor/containerexecutor/tmpl.go:215 +0x1ed │
│ github.com/kubeshop/testkube/pkg/executor/containerexecutor.(*ContainerExecutor).createJob(_, {_, _}, {{0xc001be9c68, 0x18}, {0xc0018494b8, 0x13}, {0x0, 0x0}, {0xc000eb5880, ...}, ...}, │
│ ...) │
│ /home/runner/work/testkube/testkube/pkg/executor/containerexecutor/containerexecutor.go:305 +0x4d1 │
│ github.com/kubeshop/testkube/pkg/executor/containerexecutor.(*ContainerExecutor).Execute(_, {_, _}, _, {{0xc001be9c68, 0x18}, {0xc0018494b8, 0x13}, {0xc000eb5880, 0xa}, ...}) │
│ /home/runner/work/testkube/testkube/pkg/executor/containerexecutor/containerexecutor.go:253 +0x174 │
│ github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).startTestExecution(_, {_, _}, {{0xc001be9c68, 0x18}, {0xc0018494b8, 0x13}, {0xc000eb5880, 0xa}, {{0xc001a00500, ...}, ...}, ...}, │
│ ...) │
│ /home/runner/work/testkube/testkube/pkg/scheduler/test_scheduler.go:163 +0x82 │
│ github.com/kubeshop/testkube/pkg/scheduler.(*Scheduler).executeTest(_, {_, _}, {{0xc0018494b8, 0x13}, {0xc000eb5880, 0xa}, {0x0, 0x0}, {0xc000eb5920, ...}, ...}, ...) │
│ /home/runner/work/testkube/testkube/pkg/scheduler/test_scheduler.go:104 +0x1154 │
│ github.com/kubeshop/testkube/pkg/workerpool.Request[...].execute(_, {_, _}) │
│ /home/runner/work/testkube/testkube/pkg/workerpool/service.go:43 +0xdf │
│ github.com/kubeshop/testkube/pkg/workerpool.worker[...]({0x2a50ae8, 0xc002349200}, 0xc0013eb8c0, 0xc000a874a0?, 0xc000a87500) │
│ /home/runner/work/testkube/testkube/pkg/workerpool/service.go:108 +0xd2 │
│ created by github.com/kubeshop/testkube/pkg/workerpool.Service[...].Run in goroutine 594 │
│ /home/runner/work/testkube/testkube/pkg/workerpool/service.go:77 +0x6c
```

--- a/pkg/skopeo/client.go
+++ b/pkg/skopeo/client.go
@@ -84,7 +84,7 @@ func (c *client) Inspect(image string) (*DockerImage, error) {
 	}
 
 	if len(c.dockerAuthConfigs) != 0 {
-		i := 1 + rand.Intn(len(c.dockerAuthConfigs))
+		i := rand.Intn(len(c.dockerAuthConfigs))
 		args = append(args, "--creds", c.dockerAuthConfigs[i].Username+":"+c.dockerAuthConfigs[i].Password)
 	}
 

